roachprod: don't show progress when downloading grafana

RaduBerinde opened the pull request:
When a roachtest downloads Grafana, `wget` logs a lot of spammy lines
with progress reports. In this commit, we switch to `curl` (which is
typically used in roachprod) and pass the `-sS` flags to only show errors.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/109038)
<!-- Reviewable:end -->

RaduBerinde commented:
TFTR!

bors r+

stevendanna commented on pkg/roachprod/prometheus/prometheus.go:
```diff
@@ -343,10 +343,11 @@ sudo systemd-run --unit prometheus --same-dir \
 sudo apt-get install -y adduser libfontconfig1 &&
-wget https://dl.grafana.com/enterprise/release/grafana-enterprise_9.2.3_%s.deb -O grafana-enterprise_9.2.3_%s.deb &&
-sudo dpkg -i grafana-enterprise_9.2.3_%s.deb &&
+echo "Downloading https://dl.grafana.com/enterprise/release/grafana-enterprise_9.2.3_%[1]s.deb" &&
+wget https://dl.grafana.com/enterprise/release/grafana-enterprise_9.2.3_%[1]s.deb -O grafana-enterprise_9.2.3_%[1]s.deb --no-verbose &&
```
This is fine by me. But if you wanted, I think this is the only place we use wget over curl.

RaduBerinde commented:
bors r-

RaduBerinde committed:
roachprod: don't show progress when downloading grafana

When a roachtest downloads Grafana, `wget` logs a lot of spammy lines
with progress reports. In this commit, we switch to `curl` (which is
typically used in roachprod) and pass the `-sS` flags to only show
errors.

Epic: none
Release note: None

RaduBerinde reviewed (commented):
Update to use `curl` instead.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/109038#-:-NcYezlW3-ZQp5T0FftY:bdi01d8)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @DarrylWong, @srosenberg, and @stevendanna)

___
*[`pkg/roachprod/prometheus/prometheus.go` line 347 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/109038#gh-1303386091:-NcYevFa44HJQXByTW9D:bsnpunm) ([raw file](https://github.com/cockroachdb/cockroach/blob/b3a19998425a198fe09bbd95154f0a358a044b4d/pkg/roachprod/prometheus/prometheus.go#L347)):*
<details><summary><i>Previously, stevendanna (Steven Danna) wrote…</i></summary><blockquote>

This is fine by me. But if you wanted, I think this is the only place we use wget over curl.
</blockquote></details>

Good point, fixed. Used `-sS` to disable progress bar but show errors.


<!-- Sent from Reviewable.io -->

stevendanna commented on pkg/roachprod/prometheus/prometheus.go:
```diff
@@ -341,12 +341,13 @@ sudo systemd-run --unit prometheus --same-dir \
-wget https://dl.grafana.com/enterprise/release/grafana-enterprise_9.2.3_%s.deb -O grafana-enterprise_9.2.3_%s.deb &&
-sudo dpkg -i grafana-enterprise_9.2.3_%s.deb &&
+echo "Downloading https://dl.grafana.com/enterprise/release/grafana-enterprise_9.2.3_%[1]s.deb" &&
+curl https://dl.grafana.com/enterprise/release/grafana-enterprise_9.2.3_%[1]s.deb -sS -o grafana-enterprise_9.2.3_%[1]s.deb &&
```
You may want -f to get errors on non-2xx responses and -L to follow redirects.

RaduBerinde replied:
According to my man page, -f means "Fail fast with no output at all on server errors"

stevendanna reviewed (approved):
Thanks. Noted a couple of other flags you might want to add, but I leave that to your judgement.

RaduBerinde commented:
bors r+
